### DISCUSSION TITLE
v1.117.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#13135] [Announcements]
# v1.117.0

> [!WARNING]
> ## Breaking changes
> The `image` section of the config file structure for thumbnails and previews has changed. If you use a config file and set the image settings to custom values, these will be ignored until updated to the new structure.

```diff
…
"image": {
-  “previewFormat”: “jpeg”,
-  “previewSize”: 1440,
-  “quality”: 80,
-  “thumbnailFormat”: “webp”,
-  “thumbnailSize”: 250,
+  "thumbnail": {
+     "format": "webp",
+     "size": 250,
+     "quality": 80
+   },
+   "preview": {
+     "format": "jpeg",
+     "size": 1440,
+     "quality": 80
+   },
    "colorspace": "p3",
    "extractEmbedded": false
  }
…
```

## Highlights

Welcome to release `v1.117.0` of Immich. This release continues to bring bugfixes and additional enhancements to the app. Let's go over some of the highlights below:

- Better folder checks
- Download notifications (mobile)
- Support and feedback links (web)
- Upgrade history
- Asset thumbnail improvements

### Better folder checks
This release fixes a few more issues with the `.immich` related checks.
- Ignore errors when `.immich` already exists, but mount checks aren't enabled
- Add a docs page about [System Integrity](https://immich.app/docs/administration/system-integrity)
- Add an env option to skip mount checks (`IMMICH_IGNORE_MOUNT_CHECK_ERRORS=true`)

### Download notifications (mobile)
Downloads now show their progress and can be canceled

<img src="https://github.com/user-attachments/assets/59e506be-549d-472c-bcc5-938f892dbe68" width="300px"/>

### Support and feedback links (web)
Links have been added to the Immich web application to help users more easily find our documentation, source code, discord, etc. Also, third party packages have the option to additionally include their own links.

![image](https://github.com/user-attachments/assets/b067b692-df64-466f-a838-67d25837175e)

![image](https://github.com/user-attachments/assets/1a9c5abe-529b-4cef-91f9-1470e1491dde)

### Upgrade history
Clicking on the server build number on the web will now also show upgrade history, in addition to other build information.

In the future we may look at combining this information with the new support/help modal.

| Light | Dark
| - | - |
| ![image](https://github.com/user-attachments/assets/28d0a046-d1bc-41bc-976c-65e19f559187) | ![image](https://github.com/user-attachments/assets/26e39904-e1ea-4ec3-890e-7ab8f9cca136) |

#### Combined jobs
Thumbnails are critical to keep Immich feeling snappy. We currently generate a "preview" (large), a "thumbnail" (small), and a "thumbhash" (very small) for each asset. Prior to this release each version for each asset was a separate job. Now they have been combined into a single job, per asset. This makes the jobs page much more accurate and useful. It also means thumbnails generate faster, and use less system resources (see below).

![thumbnail-generation-memory-usage](https://github.com/user-attachments/assets/b31850c5-c283-4a87-b4c0-28b05867b7f4)

In order from left to right:
  - Previews generation start on old version
  - Previews generation completed old version (starting thumbnails)
  - Thumbnail generation finished old version
  - Thumbnail generation started on new version
  - Thumbnail generation finished on new version

#### Common settings
Previews and thumbnails can now _both_ set desired `format`, `resolution`, and `quality` settings.

![image-settings](https://github.com/user-attachments/assets/589648c7-6687-411e-8a9e-f1e0a0515a88)

## Support Immich

<p align="center">
<img src="https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExbjY2eWc5Y2F0ZW56MmR4aWE0dDhzZXlidXRmYWZyajl1bWZidXZpcyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/87CKDqErVfMqY/giphy.gif" width="450" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich by purchasing a product key at [https://buy.immich.app](https://buy.immich.app).

Cheers! 🍻

- - - -

### 🚀 Features
* feat: support and feedback modal with third party support by @zackpollard in https://github.com/immich-app/immich/pull/13056
* feat: track upgrade history by @jrasm91 in https://github.com/immich-app/immich/pull/13097
### 🌟 Enhancements
* feat(server): better transcoding logs by @mertalev in https://github.com/immich-app/immich/pull/13000
* feat(server): separate quality for thumbnail and preview images by @mertalev in https://github.com/immich-app/immich/pull/13006
* feat(server): generate all thumbnails for an asset in one job by @mertalev in https://github.com/immich-app/immich/pull/13012
* feat(mobile): enhance download operations by @alextran1502 in https://github.com/immich-app/immich/pull/12973
* feat(server): parse offset from "Image_UTC_Data" (Samsung) by @C-Otto in https://github.com/immich-app/immich/pull/13080
* feat(server): better mount checks by @jrasm91 in https://github.com/immich-app/immich/pull/13092
### 🐛 Bug fixes
* fix(server): "all" button for facial recognition deleting faces instead of unassigning them by @mertalev in https://github.com/immich-app/immich/pull/13042
* fix(server): `/search/random` failing with certain options by @mertalev in https://github.com/immich-app/immich/pull/13040
* fix(mobile): share to error by @alextran1502 in https://github.com/immich-app/immich/pull/13044
* fix: deletedAt not set for offline assets during 1.116.0 migration by @zackpollard in https://github.com/immich-app/immich/pull/13086
* fix(server): offline assets don't restore when coming back online by @zackpollard in https://github.com/immich-app/immich/pull/13087
### 📚 Documentation
* docs: update config file by @mertalev in https://github.com/immich-app/immich/pull/13041
* fix(docs): Update TrueNAS installation docs by @PetrSvirak in https://github.com/immich-app/immich/pull/13017
### 🌐 Translations
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/12938
* chore: add more languages by @danieldietzler in https://github.com/immich-app/immich/pull/13127

## New Contributors
* @PetrSvirak made their first contribution in https://github.com/immich-app/immich/pull/13017

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.116.2...v1.117.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.117.0'>v1.117.0</a>.</em>

**Comment by gbcatrinoiu** [comment]
Thanks for making this even better!

> We currently generate a "preview" (large), a "thumbnail" (small), and a "thumbhash" (very small) for each asset.

Is this available in the API as well? The viewAsset endpoint has options only for `preview` and `thumbnail`.

**Reply by alextran1502** [reply]
The thumbhash content is included in the API responses `http://serverip:2283/api/assets/<asset-id>`

**Reply by gbcatrinoiu** [reply]
so the difference between `/assets/<asset-id>` and `/assets/<asset-id>/<size>` is that the first contains everything about the asset while the last contains only the file?

I am now using the `/assets/<asset-id>/<size>` in a react app and I would like to use the thumbhash size but seems not to be available. Shall I use `/assets/<asset-id>`?

**Reply by alextran1502** [reply]
I think if you want to get the thumbhash content, you first call the `/assets/<asset-id>` then get the content and start the rendering process

**Reply by jrasm91** [reply]
The thumb hash is so small it is stored in the database and returned in the json response as a base64encoded buffer

**Comment by gwensg35** [comment]
Hello everybody,
I am new to Immish and I wish to congratulate all of you for the great work.
I am now facing an issue with this 1.117 which was not noticed with 1.116.2
The date displayed in the detail of the photo is now the file creation date and not the photo shotting date anymore.
![Issue1](https://github.com/user-attachments/assets/5313c61b-f850-4b33-9446-0ec0144f11bf)
![Issue2](https://github.com/user-attachments/assets/3fedd424-f197-4cc4-b1a4-22bf9c051879)
![Issue3](https://github.com/user-attachments/assets/3508ab54-2c57-4cb0-a7f2-afaf75e98adb)

Thanks and best regards,

**Reply by alextran1502** [reply]
Can you please open a GitHub issue and check the EXIF content using exiftool

**Reply by gwensg35** [reply]
Ok, I will do that
Thanks

**Reply by C-Otto** [reply]
#13141

**Comment by Blintmester** [comment]
For those, who are almost opened an issue, due to the server isn't starting, just give it to a 5 to 10 minutes, and it'll start!
I had to comment out my readiness and liveness probes, and while I writing the issue about it, after a 10 minutes it started 🤣...
So don't worry, just wait!

**Reply by devclef** [reply]
Thanks for this, was exactly in that process.

Was this a one time issue for you, or have you permanently removed these probes? 

Oddly leaving it with probes enabled for day(s) it never completed, but removing them I agree the server now came up. 

Going to try and find the change for this(did they change the probes?), otherwise feels like an issue if readiness/health probes are no longer functioning?

**Reply by mertalev** [reply]
Immich replaces geodata after updating the version. Database migrations added across releases can also take some time if the library is big. These are part of bootstrapping and Immich won't start accepting requests until they're done.

These are both "one-time" in the sense that restarting immich should be much quicker after that, but updating immich later can have the same effect.

**Reply by ilopmar** [reply]
The same happened to me and I was about to rollback the version. It would be nice to add a message to the logs so we know what's going on. Something like `Please wait while applying database migrations. It may take some time...` good be enough.

**Reply by devclef** [reply]
For others -- did it actually complete with the checks enabled? 

I semi auto-update immich to newer tags(Yes risky incase of breaking changes:)). Did that as normal, but later noticed immich was down.  Went to investigate. The workload was running, but not ready. Checking logs did not have errors but just sort of stopped logging anything after a couple non-error messages(forgot the exact message). I was sick so actually left immich down for a couple days. 

I too was about to revert when I saw this comment. So I then commented out the two probes, and then things came up as expected. 

These database migrations, do they happen entirely within the running container of immich server? or have any dependencies to say the machine learning workloads? I am wondering if the workload is not marked 'ready' it will not receive any ingress. Which means if these migrations require any sort of back'n'forth via the ingress they would hang? Just trying to think what would explain why the migration seemingly did not occur with the probes enabled but did one they were removed.

I've now restored the probes and immich came up as expected.  -- If I had reproduced the hang i'd make a ticket.

I am deploying immich to a rke2 k8s cluster using fleet. This was upgrading from  1.116.2 to 1.117.0 . 

If this happens again i'll leave it in that state and try to figure out why.  But if others had this 'just work' with probes that'll confuse me even more :)

I am doing a readiness and liveness on  /server-info/ping for the immich-server container.

**Comment by samluxh** [comment]
Big thank you for your effort ❤️
Please add a option to be able to search for pictures of a specific year or month without the full date. Example: August 2022 or August (all years) or just 2022 (all months and days).

**Comment by f-mc2** [comment]
First of all, thanks for all the work on this beautiful software!

Now, to my silly question. I do not understand where the "image section of the config file structure for thumbnails" is located, nor what it is supposed to do.  Can someone help me understand what it is?

Anyway, even if I did not apply the modifications suggested in the release note, I was able to upgrade without any (apparent) problem.

**Reply by bo0tzz** [reply]
If you're not using a config file, no need to worry about it.

**Reply by f-mc2** [reply]
I see. It actually makes a lot of sense. Thanks for the quick reply.

**Comment by imohitkr** [comment]
Hi, 
thank you for the update but seems the smart search job is broken.  

I am running the open vino image and everything works fine with the image built with the tag "ghcr.io/immich-app/immich-machine-learning:pr-11453-openvino" but the latest image with  tag "ghcr.io/immich-app/immich-machine-learning:v1.117.0-openvino",  I am getting the errors tagged below.  When i pair the latest version with the old machine learning image everything works fine. 

Interestingly the "text" based search works ,where it loads the textual model , with the latest machine-learning-server image as well. 

Please help. 

 

<details>
<pre>

[10/06/24 15:11:46] INFO     Attempt #2 to load visual model                    

                             'ViT-H-14-378-quickgelu__dfn5b' to memory          

[10/06/24 15:11:46] INFO     Setting execution providers to                     

                             ['OpenVINOExecutionProvider',                      

                             'CPUExecutionProvider'], in descending order of    

                             preference                                         

2024-10-06 15:11:46.370454587 [E:onnxruntime:, inference_session.cc:2105 operator()] Exception during initialization: /onnxruntime/onnxruntime/core/providers/openvino/backend_manager.cc:126 onnxruntime::openvino_ep::BackendManager::BackendManager(const onnxruntime::openvino_ep::GlobalContext&, const onnxruntime::Node&, const onnxruntime::GraphViewer&, const onnxruntime::logging::Logger&, onnxruntime::openvino_ep::EPCtxHandler&) /onnxruntime/onnxruntime/core/providers/openvino/ov_interface.cc:106 onnxruntime::openvino_ep::OVExeNetwork onnxruntime::openvino_ep::OVCore::CompileModel(const std::string&, std::string&, ov::AnyMap&, const std::string&) [OpenVINO-EP]  Exception while Loading Network for graph: OpenVINOExecutionProvider_OpenVINO-EP-subgraph_2_0Exception from src/inference/src/cpp/core.cpp:142:

invalid external data: ExternalDataInfo(data_full_path: 51cc7752-47e9-11ef-91e8-00155d655292, offset: 9564160, data_length: 3737600)

[10/06/24 15:11:46] ERROR    Exception in ASGI application                      

                                                                                

                             ╭─────── Traceback (most recent call last) ───────╮

                             │ /usr/src/app/main.py:152 in predict             │

                             │                                                 │

                             │   149 │   │   inputs = text                     │

                             │   150 │   else:                                 │

                             │   151 │   │   raise HTTPException(400, "Either  │

                             │ ❱ 152 │   response = await run_inference(inputs │

                             │   153 │   return ORJSONResponse(response)       │

                             │   154                                           │

                             │   155                                           │

                             │                                                 │

                             │ /usr/src/app/main.py:175 in run_inference       │

                             │                                                 │

                             │   172 │   │   response[entry["task"]] = output  │

                             │   173 │                                         │

                             │   174 │   without_deps, with_deps = entries     │

                             │ ❱ 175 │   await asyncio.gather(*[_run_inference │

                             │   176 │   if with_deps:                         │

                             │   177 │   │   await asyncio.gather(*[_run_infer │

                             │   178 │   if isinstance(payload, Image):        │

                             │                                                 │

                             │ /usr/src/app/main.py:169 in _run_inference      │

                             │                                                 │

                             │   166 │   │   │   except KeyError:              │

                             │   167 │   │   │   │   message = f"Task {entry[' │

                             │       output of {dep}"                          │

                             │   168 │   │   │   │   raise HTTPException(400,  │

                             │ ❱ 169 │   │   model = await load(model)         │

                             │   170 │   │   output = await run(model.predict, │

                             │   171 │   │   outputs[model.identity] = output  │

                             │   172 │   │   response[entry["task"]] = output  │

                             │                                                 │

                             │ /usr/src/app/main.py:213 in load                │

                             │                                                 │

                             │   210 │   │   return model                      │

                             │   211 │                                         │

                             │   212 │   try:                                  │

                             │ ❱ 213 │   │   return await run(_load, model)    │

                             │   214 │   except (OSError, InvalidProtobuf, Bad │

                             │   215 │   │   log.warning(f"Failed to load {mod │

                             │       '{model.model_name}'. Clearing cache.")   │

                             │   216 │   │   model.clear_cache()               │

                             │                                                 │

                             │ /usr/src/app/main.py:188 in run                 │

                             │                                                 │

                             │   185 │   if thread_pool is None:               │

                             │   186 │   │   return func(*args, **kwargs)      │

                             │   187 │   partial_func = partial(func, *args, * │

                             │ ❱ 188 │   return await asyncio.get_running_loop │

                             │   189                                           │

                             │   190                                           │

                             │   191 async def load(model: InferenceModel) ->  │

                             │                                                 │

                             │ /usr/local/lib/python3.11/concurrent/futures/th │

                             │ read.py:58 in run                               │

                             │                                                 │

                             │ /usr/src/app/main.py:200 in _load               │

                             │                                                 │

                             │   197 │   │   │   raise HTTPException(500, f"Fa │

                             │   198 │   │   with lock:                        │

                             │   199 │   │   │   try:                          │

                             │ ❱ 200 │   │   │   │   model.load()              │

                             │   201 │   │   │   except FileNotFoundError as e │

                             │   202 │   │   │   │   if model.model_format ==  │

                             │   203 │   │   │   │   │   raise e               │

                             │                                                 │

                             │ /usr/src/app/models/base.py:53 in load          │

                             │                                                 │

                             │    50 │   │   self.download()                   │

                             │    51 │   │   attempt = f"Attempt #{self.load_a │

                             │       else "Loading"                            │

                             │    52 │   │   log.info(f"{attempt} {self.model_ │

                             │       '{self.model_name}' to memory")           │

                             │ ❱  53 │   │   self.session = self._load()       │

                             │    54 │   │   self.loaded = True                │

                             │    55 │                                         │

                             │    56 │   def predict(self, *inputs: Any, **mod │

                             │                                                 │

                             │ /usr/src/app/models/clip/visual.py:62 in _load  │

                             │                                                 │

                             │   59 │   │   self.mean = np.array(self.preproce │

                             │   60 │   │   self.std = np.array(self.preproces │

                             │   61 │   │                                      │

                             │ ❱ 62 │   │   return super()._load()             │

                             │   63 │                                          │

                             │   64 │   def transform(self, image: Image.Image │

                             │   65 │   │   image = resize_pil(image, self.siz │

                             │                                                 │

                             │ /usr/src/app/models/base.py:78 in _load         │

                             │                                                 │

                             │    75 │   │   )                                 │

                             │    76 │                                         │

                             │    77 │   def _load(self) -> ModelSession:      │

                             │ ❱  78 │   │   return self._make_session(self.mo │

                             │    79 │                                         │

                             │    80 │   def clear_cache(self) -> None:        │

                             │    81 │   │   if not self.cache_dir.exists():   │

                             │                                                 │

                             │ /usr/src/app/models/base.py:110 in              │

                             │ _make_session                                   │

                             │                                                 │

                             │   107 │   │   │   case ".armnn":                │

                             │   108 │   │   │   │   session: ModelSession = A │

                             │   109 │   │   │   case ".onnx":                 │

                             │ ❱ 110 │   │   │   │   session = OrtSession(mode │

                             │   111 │   │   │   case _:                       │

                             │   112 │   │   │   │   raise ValueError(f"Unsupp │

                             │   113 │   │   return session                    │

                             │                                                 │

                             │ /usr/src/app/sessions/ort.py:28 in __init__     │

                             │                                                 │

                             │    25 │   │   self.providers = providers if pro │

                             │    26 │   │   self.provider_options = provider_ │

                             │       self._provider_options_default            │

                             │    27 │   │   self.sess_options = sess_options  │

                             │       self._sess_options_default                │

                             │ ❱  28 │   │   self.session = ort.InferenceSessi │

                             │    29 │   │   │   self.model_path.as_posix(),   │

                             │    30 │   │   │   providers=self.providers,     │

                             │    31 │   │   │   provider_options=self.provide │

                             │                                                 │

                             │ /opt/venv/lib/python3.11/site-packages/onnxrunt │

                             │ ime/capi/onnxruntime_inference_collection.py:41 │

                             │ 9 in __init__                                   │

                             │                                                 │

                             │    416 │   │   disabled_optimizers = kwargs.get │

                             │    417 │   │                                    │

                             │    418 │   │   try:                             │

                             │ ❱  419 │   │   │   self._create_inference_sessi │

                             │        disabled_optimizers)                     │

                             │    420 │   │   except (ValueError, RuntimeError │

                             │    421 │   │   │   if self._enable_fallback:    │

                             │    422 │   │   │   │   try:                     │

                             │                                                 │

                             │ /opt/venv/lib/python3.11/site-packages/onnxrunt │

                             │ ime/capi/onnxruntime_inference_collection.py:49 │

                             │ 1 in _create_inference_session                  │

                             │                                                 │

                             │    488 │   │   │   disabled_optimizers = set(di │

                             │    489 │   │                                    │

                             │    490 │   │   # initialize the C++ InferenceSe │

                             │ ❱  491 │   │   sess.initialize_session(provider │

                             │    492 │   │                                    │

                             │    493 │   │   self._sess = sess                │

                             │    494 │   │   self._sess_options = self._sess. │

                             ╰─────────────────────────────────────────────────╯

                             RuntimeException: [ONNXRuntimeError] : 6 :         

                             RUNTIME_EXCEPTION : Exception during               

                             initialization:                                    

                             /onnxruntime/onnxruntime/core/providers/openvino/ba

                             ckend_manager.cc:126                               

                             onnxruntime::openvino_ep::BackendManager::BackendMa

                             nager(const                                        

                             onnxruntime::openvino_ep::GlobalContext&, const    

                             onnxruntime::Node&, const                          

                             onnxruntime::GraphViewer&, const                   

                             onnxruntime::logging::Logger&,                     

                             onnxruntime::openvino_ep::EPCtxHandler&)           

                             /onnxruntime/onnxruntime/core/providers/openvino/ov

                             _interface.cc:106                                  

                             onnxruntime::openvino_ep::OVExeNetwork             

                             onnxruntime::openvino_ep::OVCore::CompileModel(cons

                             t std::string&, std::string&, ov::AnyMap&, const   

                             std::string&) [OpenVINO-EP]  Exception while       

                             Loading Network for graph:                         

                             OpenVINOExecutionProvider_OpenVINO-EP-subgraph_2_0E

                             xception from src/inference/src/cpp/core.cpp:142:  

                             invalid external data:                             

                             ExternalDataInfo(data_full_path:                   

                             51cc7752-47e9-11ef-91e8-00155d655292, offset:      

                             9564160, data_length: 3737600) 
</pre>
<\details>

**Reply by C-Otto** [reply]
Could you open a GitHub issue for this please?

**Reply by imohitkr** [reply]
> Could you open a GitHub issue for this please?

opened here : https://github.com/immich-app/immich/issues/13227

**Comment by IoSonoAndreaZ** [comment]
Sorry, where is this "image" section to edit?

**Reply by bo0tzz** [reply]
It's only relevant if you're using a config file.
